src/stores/FileBrowserStore/FileBrowserStore.ts

```diff
@@ -317,6 +317,57 @@ export class FileBrowserStore {
         return {file: res.fileInfo.name, info: res.fileInfoExtended};
     };
 
+    /**
+     * Obtains the Stokes type of the file from the header information or the filename.
+     *
+     * @param fileInfoExtended - The header information of the file.
+     * @param file - The name of the file.
+     * @returns The Stokes type of the file.
+     */
+    getStokesType = (fileInfoExtended: CARTA.IFileInfoExtended, file: string): CARTA.PolarizationType => {
+        let type = FileBrowserStore.GetTypeFromHeader(fileInfoExtended?.headerEntries);
+        if (type === CARTA.PolarizationType.POLARIZATION_TYPE_NONE) {
+            type = FileBrowserStore.GetTypeFromName(file);
+        }
+        return type;
+    };
+
+    private static GetTypeFromHeader = (headers: CARTA.IHeaderEntry[]): CARTA.PolarizationType => {
+        let type = CARTA.PolarizationType.POLARIZATION_TYPE_NONE;
+
+        const ctype = headers?.find(obj => obj.value.toUpperCase() === "STOKES");
+        if (ctype && ctype.name.indexOf("CTYPE") !== -1) {
+            const index = ctype.name.substring(5);
+            const crpixHeader = headers.find(entry => entry.name.indexOf(`CRPIX${index}`) !== -1);
+            const crvalHeader = headers.find(entry => entry.name.indexOf(`CRVAL${index}`) !== -1);
+            const cdeltHeader = headers.find(entry => entry.name.indexOf(`CDELT${index}`) !== -1);
+            const polarizationIndex = getHeaderNumericValue(crvalHeader) + (1 - getHeaderNumericValue(crpixHeader)) * getHeaderNumericValue(cdeltHeader);
+            if (polarizationIndex) {
+                const polarizationString = STANDARD_POLARIZATIONS.get(polarizationIndex);
+                if (polarizationString) {
+                    type = CARTA.PolarizationType[polarizationString] ?? CARTA.PolarizationType.POLARIZATION_TYPE_NONE;
+                }
+            }
+        }
+
+        return type;
+    };
+
+    private static GetTypeFromName = (fileName: string): CARTA.PolarizationType => {
+        let type = CARTA.PolarizationType.POLARIZATION_TYPE_NONE;
+        const separators = [".", "_"];
+        separators.forEach(separator => {
+            const words = fileName.split(separator);
+            words.forEach(word => {
+                const matchedType = CARTA.PolarizationType[word];
```

Comment:
I would suggest uppercasing `word` here, so that `i`, `q`, etc., are not ignored. Unless we think that lowercase letters are likely to have a different meaning in filenames.

Reply:
This is added.